### PR TITLE
Note controller implementation

### PR DESCRIPTION
**Pull request by 4rgc**
Implements the note controller, used to work with note entities.
API methods:

- GET /note/:boardId/:listId/:noteId – get a note
   Returns a complete note entity
- POST /note/:boardId/:listId/:noteId – update data of a note
   Allows to update note's name, description, start date, due date, and associated tags.
- POST /note/:boardId/:listId – add a note
   Add a note to a list with a required name, and other optional properties.
- DELETE /note/:boardId/:listId/:noteId - delete a note
   Deletes a note by its id

### CODE REVIEW

**Comment by 4rgc** [comment]
@stalkermn Please review 🙃

File: api/src/controllers/note.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		.getNote(userId, boardId, listId, noteId)
+		.then(
+			(note) => ({ note, err: undefined }),
+			(err) => ({ err, note: undefined })
+		);
+	if (err) return next(err);
+	if (!note) return res.status(404).json({ message: 'Note not found' });
+
+	res.json({
+		note,
```
Comment (ctrldrift) [inline api/src/controllers/note.ts:24]: isn't a note is already an object?

File: api/src/controllers/note.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	});
+};
+
+const updateNote = async (req: Request, res: Response, next: NextFunction) => {
+	const { userId } = res.locals.auth;
+	const { boardId, listId, noteId } = req.params;
+	const { name, description, startDate, dueDate, tagIds } = req.body;
+
+	if (!boardId) return res.status(400).json({ message: 'boardId was null' });
```
Comment (ctrldrift) [inline api/src/controllers/note.ts:33]: same validation code as for GET/POST? Don't you want to generalize this?

Comment (ctrldrift) [reply api/src/controllers/note.ts:33]: Example: once the param name is been changed, you will have to modify all the places where it's in use, instead a single update

File: api/src/controllers/note.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	res.status(201).json({
+		note: newNote,
+	});
+};
+
+const deleteNote = async (req: Request, res: Response, next: NextFunction) => {
+	const { userId } = res.locals.auth;
+	const { boardId, listId, noteId } = req.params;
+
+	if (!boardId) return res.status(400).json({ message: 'boardId was null' });
```
Comment (ctrldrift) [inline api/src/controllers/note.ts:103]: same input validations here

File: api/src/data-controllers/note.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+					comments: 1,
+					tags: 1,
+					checklistOrder: 1,
+				},
+			},
+		},
+	})
+		.lean()
+		.exec()
+		.then((user) => user?.boards[0].lists[0].notes[0]);
```
Comment (ctrldrift) [inline api/src/data-controllers/note.ts:31]: hope it's unique :)

**Pull request merged**